Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -790,4 +791,26 @@ void commit_a_transaction(JS::Realm& realm, GC::Ref<IDBTransaction> transaction)
     }));
 }
 
+// https://w3c.github.io/IndexedDB/#clone
+WebIDL::ExceptionOr<JS::Value> clone_in_realm(JS::Realm& target_realm, JS::Value value, GC::Ref<IDBTransaction> transaction)
+{
+    // 1. Assert: transaction’s state is active.
+    VERIFY(transaction->state() == IDBTransaction::TransactionState::Active);
+
+    // 2. Set transaction’s state to inactive.
+    transaction->set_state(IDBTransaction::TransactionState::Inactive);
+
+    // 3. Let serialized be ? StructuredSerializeForStorage(value).
+    auto serialized = TRY(HTML::structured_serialize_for_storage(target_realm.vm(), value));
```

Comment:
typically we slap a `auto& vm = someones.realm();` at the top of methods when they're used more than once.